folding-schemes/src/frontend/circom/test_folder/keccak256-circom/circuits/keccak.circom

```diff
@@ -0,0 +1,186 @@
+// Keccak256 hash function (ethereum version).
+// For LICENSE check https://github.com/vocdoni/keccak256-circom/blob/master/LICENSE
```

Comment:
Maybe we could use a simpler circuit that does not require pasting here the entire circuits of the https://github.com/vocdoni/keccak256-circom repo?

Ideally not even importing external projects, but just a self-contained & simpler circuit with few lines of code that fails with the version of the frontend/circom before this PR.

For example, in the `external_inputs.circom`, by using a template like `IsZero`:
```circom
pragma circom 2.0.3;

// This template is from circomlib/circuits/comparators.circom
template IsZero() {
    signal input in;
    signal output out;

    signal inv;

    inv <-- in!=0 ? 1/in : 0;

    out <== -in*inv +1;
    in*out === 0;
}


/*
    z_{i+1} == z_i^3 + z_i * external_input[0] + external_input[1]
*/
template Example () {
    signal input ivc_input[1]; // IVC state
    signal input external_inputs[2]; // not state

    signal output ivc_output[1]; // next IVC state

    component check_input = IsZero(); // this is what triggers the error prior to this PR
    check_input.in <== ivc_input[0];
    check_input.out === 0;

    signal temp1;
    signal temp2;
    
    temp1 <== ivc_input[0] * ivc_input[0];
    temp2 <== ivc_input[0] * external_inputs[0];
    ivc_output[0] <== temp1 * ivc_input[0] + temp2 + external_inputs[1];
}

component main {public [ivc_input]} = Example();
```
this will already fail with the version prior to this PR and could be used to test the fixes from this PR on the Circom frontend. Maybe with some other edge cases that were triggering the error prior to the PR.

Reply:
Yes, the keccak circuit, though covers everything we need, is not very practical to test. I replaced it with more lightweight circuits. 
We still depend on one copied-pasted `circomlib` circuit (your suggested`IsZero` template): the test will re-use it for two different cases, namely with and without external inputs. Tests will also check that circom defined constraints are well enforced by feeding in erroneous inputs.